ndcube/asdf/converters/globalcoords_converter.py

```diff
@@ -0,0 +1,28 @@
+from typing import OrderedDict
+
+from asdf.extension import Converter
+
+
+class GlobalCoordsConverter(Converter):
+    tags = ["tag:sunpy.org:ndcube/global_coords/GlobalCoords-*"]
+    types = ["ndcube.global_coords.GlobalCoords"]
+
+    def from_yaml_tree(self, node, tag, ctx):
+        from ndcube.global_coords import GlobalCoords
+
+        globalcoords = GlobalCoords()
+        if node.get("internal_coords") is not None:
+            globalcoords._internal_coords = OrderedDict(node.get("internal_coords"))
+        globalcoords._ndcube = node["ndcube"]
+
+        return globalcoords
+
+    def to_yaml_tree(self, globalcoords, tag, ctx):
+        node = {}
+        node["ndcube"] = globalcoords._ndcube
+        if globalcoords._internal_coords:
+            node["internal_coords"] = dict(globalcoords._internal_coords)
+        # Todo: Include `_all_coords` as a node key to preserve the dropped dimensions
```

Comment:
I don't think we do need to include all_coords, the ndcube object should have the same `SlicedLowLevelWCS` so the rest of `all_coords` will be dynamically generated as it is before we save it.

Reply:
Ah, I see. Since I couldn't test it with the sliced ndcube, I was skeptical about whether to add all_coords or not.